src/mna.hrl

```diff
@@ -0,0 +1,4 @@
+-record(mna, {
+    module = undefined :: atom() | undefined,
+    name               :: atom(),
+    arity  = undefined :: integer() | undefined}).
```

Comment:
By unifying the errors for undefined local and remote functions, records and types, the module and arity becomes optional in all of these type errors. That's expanding the type by allowing a lot more cases which don't exist in reality. For some errors we always have an arity and in other cases we don't. Therefore, I don't like this approach.

Reply:
To me it's worse that the formatting of M:N/A is spread out in several cases since that both makes our code more verbose and it makes for an API that is more difficult to use. We can always add an intersection type when time comes that would remove the impossible combinations.  That's how I would do it in Agda for example - a fairly simple type combined with propositions that restrict it to make it exact.

Reply:
I'll move this commit to another PR if you still don't agree.

Reply:
Good idea. :-)

Looking at the types, it means anyone implementing this will need to be able to handle records like `mod:#rec{}/3`.

The error message where we refer to `mod:#rec{}` isn't even correct Erlang syntax, so we should probably write this as "`#rec{}` in the `mod` module" instead.

Another way is to represent the remote module as the file annotation in the type, e.g. as `{undef_record, `{record, [{location, {3,5}}, {file, "mod.erl"}], rec}}`. That's how gradualizer_db returns them anyway.

The interface for these type errors is quite large anyway, so I think a well-documented interface on this form would be hard to maintain. Instead, we can have an opaque error type and they can supply pretty-printing funs for expressions an types. That's a very clear interface IMO.